tests/integration/hive_metastore/test_catalog_schema.py

```diff
@@ -3,21 +3,37 @@
 
 import pytest
 from databricks.labs.blueprint.tui import MockPrompts
+from databricks.sdk import WorkspaceClient
 from databricks.sdk.errors import NotFound
 from databricks.sdk.retries import retried
-from databricks.sdk.service.catalog import CatalogInfo
+from databricks.sdk.service.catalog import CatalogInfo, PermissionsList, SchemaInfo
 from databricks.sdk.service.compute import DataSecurityMode, AwsAttributes
 from databricks.sdk.service.catalog import Privilege, SecurableType, PrivilegeAssignment
 from databricks.sdk.service.iam import PermissionLevel
 
-from databricks.labs.ucx.hive_metastore.grants import GrantsCrawler
+from databricks.labs.ucx.hive_metastore.grants import Grant
 from databricks.labs.ucx.hive_metastore.mapping import Rule
 from ..conftest import get_azure_spark_conf
 
 logger = logging.getLogger(__name__)
 _SPARK_CONF = get_azure_spark_conf()
 
 
+@retried(on=[NotFound], timeout=timedelta(seconds=20))
+def get_catalog(ws: WorkspaceClient, name: str) -> CatalogInfo:
+    return ws.catalogs.get(name)
+
+
+@retried(on=[NotFound], timeout=timedelta(seconds=20))
+def get_schema(ws: WorkspaceClient, full_name: str) -> SchemaInfo:
+    return ws.schemas.get(full_name)
+
+
+@retried(on=[NotFound], timeout=timedelta(seconds=20))
+def get_schema_permissions_list(ws: WorkspaceClient, full_name: str) -> PermissionsList:
+    return ws.grants.get(SecurableType.SCHEMA, full_name)
```

Comment:
move this logic down to `CatalogSchema` class, so that you handle eventual consistency in production code, not in the test code. remember - CREATE-READ-READ is the pattern used in Terraform to handle these situations.

e.g. this makes the wrong thing more robust :)

Reply:
See the changes to the `CatalogSchema` class. 

Note that I went for a `(READ-)CREATE-READ` pattern:
- `READ` before `CREATE` to check if resource already exists.
- One `READ` after create because all settings can be set during the `CREATE` which is not always the case for Terraform resources. 

Also note that I kept `get_schema_permissions_list` as this uses a different API for `CREATE` (with SQL) and `READ` (with SDK/RestAPI). For now, I do not think it is worth it to implement a mapping between these two APIs to introduce the `CREATE-READ` pattern.